alica_engine/include/engine/BasicPlan.h

```diff
@@ -28,7 +28,7 @@ class BasicPlan : private RunnableObject
     virtual void onInit(){};
     virtual void run(void* msg){};
     virtual void onTerminate(){};
-    virtual std::unordered_map<int64_t, std::unordered_set> getApplicationEntrypointContext(vector<int64_t> entryPointIds){};
+    virtual const std::unordered_map<int64_t, std::vector<int64_t>>& getApplicationEntrypointContext(std::vector<int64_t> entryPointIds){};
```

Comment:
sorry in my my previous comment i only suggested to add std:: in front of vector in argument. The return value could still be unordered_set<int64_t>. More specifically: const std::unordered_map<int64_t, std::unordered_set<int64_t>>&